Sources/Outputs/VideoOutput/VideoOutput.State.swift

```diff
@@ -107,11 +107,12 @@ public enum VideoOutputState: Equatable {
          .preparing:
       return .preparing
 
-    case .recording(let time):
-      return .recording(time: time)
+    case let .recording(time, pause, resume):
+      return .recording(time: time, pause: pause, resume: resume)
 
-    case .paused:
-      return .preparing
+    case let .paused(time, pause):
+        let resume = CMTime(seconds: CACurrentMediaTime(), preferredTimescale: pause.timescale)
+        return .recording(time: time, pause: pause, resume: resume)
```

Comment:
It feels like CACurrentMediaTime might be a bit inaccurate and I would make this "preparing" and use the next received buffer time as the current. With CACurrentMediaTime there still can be a few extra milliseconds gap between frames.

But if it works for you I'm ok to merge this as is.

Reply:
Yes I know that it is not perfect, to get the exact time the logic to pause/resume should be done in the input so until we can avoid render a frame which is not going to be used.